app/move/controllers/create/base.js

```diff
@@ -11,13 +12,22 @@ class CreateBaseController extends FormWizardController {
     super.middlewareLocals()
     this.use(this.setCancelUrl)
     this.use(this.setMoveSummary)
+    this.use(this.setJourneyTimer)
   }
 
   setCancelUrl(req, res, next) {
     res.locals.cancelUrl = res.locals.MOVES_URL
     next()
   }
 
+  setJourneyTimer(req, res, next) {
+    if (!get(req, 'session.createMoveJourneyTimestamp')) {
+      set(req, 'session.createMoveJourneyTimestamp', new Date().getTime())
```

Comment:
Depending on our browser support, could we use `performance.now` here & for the final calculation ([caniuse](https://caniuse.com/#feat=mdn-api_performance))? `performance.now` is guaranteed to increase at a constant rate, whereas `Date().getTime()` is affected by user clock changes, NTP, etc., but it's not supported on IE8.

Reply:
This is server side google analytics.

Reply:
:+1:, ignore the browser comments. The problem I point out still stands though.

A quick Google suggests that the Performance Timing API was added to Node 8.5.0.

If you don't like that solution, I'd suggest something like `process.hrtime()`, which was added in 0.7.6, and (from [the docs](https://nodejs.org/api/process.html#process_process_hrtime_time)):

> These times are relative to an arbitrary time in the past, and not related to the time of day and therefore not subject to clock drift. The primary use is for measuring performance between intervals

Reply:
> The problem I point out still stands though.

I don't feel this is a problem TBH. It feels like this is personal preference. I'd prefer to just use `Date` and timestamps as I always have.

> whereas Date().getTime() is affected by user clock changes, NTP, etc

Timing wise it would be like for like so even if the clock changes it won't be a problem

Reply:
Hi @feedmypixel, I'm afraid this isn't true. Using `Date().getTime()` the time might change locally (e.g. through NTP) *between* the first and second reading, so the elapsed time is unreliable. The purpose of `hrtime()` is to prevent that, and it guarantees it will increment predictably over time.

The Node documentation linked above cites performance benchmarking as the primary use of `process.hrtime()`. This is a great opportunity to learn new approaches, and since I learned about the unreliability of local clock time I've applied this across all the projects I've contributed to.

Is there a compelling reason *not* to use `process.hrtime()`?

Reply:
> Is there a compelling reason _not_ to use `process.hrtime()`?

We are measuring the time of a journey in seconds. What you are talking about is a real edge case. If we were measuring process in less than a millisecond and needed that level of accuracy then I may consider this. But for GA and journey times I feel we do not need this.

> the time might change locally (e.g. through NTP) between the first and second reading,

This is unlikely

@spikeheap I feel we are going round in circles here and this has become a personal preference thing. This is not the point of code review IMO.

If you would like to speak about this in person please give me a slack call and we can work something out.